### PR TITLE
feat: publish & delete assessment from assessments table

### PR DESCRIPTION
**Pull request by joyce-shi**
## Notion ticket link

[Delete Assessment in Display page](https://www.notion.so/uwblueprintexecs/Delete-Assessment-in-Display-page-ac139b0ba4594b0f8772a1cda795ee27?pvs=4)
[Publish Assessment in Display page](https://www.notion.so/uwblueprintexecs/Publish-Assessment-in-Display-page-46d509befc7a48b6a11fad52e66b4afe?pvs=4)
[Create generic status popover](https://www.notion.so/uwblueprintexecs/Create-generic-status-popover-0e59d5857bf74ed1a7cdfb82614fb598?pvs=4)

## Implementation description
* fixed bug in `publishTest` resolver
* fixed [bug in users page](https://www.notion.so/uwblueprintexecs/Clicking-outside-popover-should-close-it-dbaae46fe31945e58aadcf8555b62dc7?pvs=4)
* added publish assessment button & modal to display assessments page
* added delete assessment button & modal to display assessments page
* created toast component to show when assessment has been published / deleted successfully (or unsuccessfully)

Note: the position and styling of the `EditStatusPopover` is slightly off and `EditStatusButton` can probably be refactored into a theme variant instead of a component. however, this isn't super high priority, so i created a ticket in the backlog for this work

## Steps to test
1. test on staging

## What should reviewers focus on?
* publish assessment works as expected
* delete assessment works as expected

### CODE REVIEW

File: frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
   TabPanels,
   Tabs,
   Text,
+  useToast,
```
Comment (carissa-tang) [inline frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx]:
```suggestion
```

File: frontend/src/components/assessments/EditStatusPopover.tsx [hunk 2]
```diff
@@ -45,7 +48,21 @@ const EditStatusPopover = (): React.ReactElement => {
...
+          <VStack spacing="0em">
+            {assessmentStatus === Status.DRAFT && (
+              <>
+                <PublishAssessmentButton
+                  assessmentId={assessmentId}
+                  closePopover={onClose}
+                />
+                <Divider borderColor="grey.200" px="17%" />
+              </>
+            )}
```
Comment (carissa-tang) [inline frontend/src/components/assessments/EditStatusPopover.tsx]:
```suggestion
          <VStack divider={<StackDivider borderColor="grey.200" />} spacing="0em">
            {assessmentStatus === Status.DRAFT && (
                <PublishAssessmentButton
                  assessmentId={assessmentId}
                  closePopover={onClose}
                />
            )}
```

File: frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx [hunk 3]
```diff
@@ -85,6 +87,7 @@ const DisplayAssessmentsPage = (): React.ReactElement => {
       filteredTests = filteredTests.filter(
         (assessment: AssessmentProperties) => {
           const assessmentProperties = [
+            assessment.id,
```
Comment (carissa-tang) [inline frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx]: can you refactor lines 214, 220, and 226 to use the enum? also i think the tabs don't show the assessments correctly?

image

Comment (joyce-shi) [reply frontend/src/components/pages/admin/DisplayAssessmentsPage.tsx]: oops. should be resolved now!

File: frontend/src/themes/components/button.ts [hunk 4]
```diff
@@ -136,6 +136,40 @@ const Button = {
...
+          bg: "red.100",
+          border: "none",
+        },
+      },
+      WhiteIcon,
+    },
+    deleteCancel: {
+      bg: "red.50",
+      color: "red.200",
+      _active: {
```
Comment (carissa-tang) [inline frontend/src/themes/components/button.ts:159]: maybe we can factor out the active and hover styling into a constant

**Review by carissa-tang — APPROVED** [review]

File: frontend/src/themes/components/button.ts (hunk 4 above)
Comment (joyce-shi) [reply frontend/src/themes/components/button.ts:159]: good idea! im gonna do that in my next pr though cuz im blocking like 3 people with this pr 😅

**Pull request merged**